main.ml

```diff
@@ -420,7 +421,8 @@ module Attached_SRs = struct
   let state_path = ref None
 
   let add smapiv2 plugin uids =
-    Hashtbl.set !sr_table ~key:smapiv2 ~data:{ sr = plugin; uids };
+    let key = Storage_interface.Sr.string_of smapiv2 in
+    Hashtbl.set !sr_table ~key ~data:{ sr = plugin; uids };
```

Comment:
It's a `Core` Hashtbl rather than `Stdlib`, and appears to be more fussy about the sorts of keys it accepts. It probably requires it to have a non-polymorphic compare perhaps?

Reply:
There is an example here, looks like it wants quite a few more things to be defined about its key: https://ocaml.janestreet.com/ocaml-core/108.08.00//doc/core/Hashtbl.html
There is a Hashtbl.Poly also